metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -55,25 +60,37 @@ public class FeatureControlManager {
      */
     private final TimelineHashMap<String, Short> finalizedVersions;
 
+    /**
+     * The current metadata version
+     */
+    private final TimelineObject<MetadataVersion> metadataVersion;
+
+    /**
+     * Collection of listeners for when features change
+     */
+    private final Map<String, FeatureLevelListener> listeners;
 
     FeatureControlManager(LogContext logContext,
                           QuorumFeatures quorumFeatures,
                           SnapshotRegistry snapshotRegistry) {
         this.log = logContext.logger(FeatureControlManager.class);
         this.quorumFeatures = quorumFeatures;
         this.finalizedVersions = new TimelineHashMap<>(snapshotRegistry, 0);
+        this.metadataVersion = new TimelineObject<>(snapshotRegistry, MetadataVersion.UNINITIALIZED);
```

Comment:
UNINITIALIZED was added since we need a non-null object to initialize the TimelineObject